src/Augurk.MSBuild/PublishFeatures.cs

```diff
@@ -236,5 +259,40 @@ private string GetFeatureUri(string groupUri, string featureTitle)
                                  groupUri,
                                  featureTitle);
         }
+
+        private HttpClient CreateHttpClient()
+        {
+            if (UseIntegratedSecurity)
+            {
+                var username = System.Security.Principal.WindowsIdentity.GetCurrent().Name;
+                Log.LogMessage($"Using integrated security with user {username} to access the Augurk API's.");
+
+                var handler = new HttpClientHandler
+                {
+                    UseDefaultCredentials = true,
+                    PreAuthenticate = true,
+                };
+                var secureClient = new HttpClient(handler);
+                return secureClient;
+            }
+
+            var client = new HttpClient();
+
+            if (UseBasicAuthentication)
+            {
+                if (string.IsNullOrEmpty(BasicAuthenticationUsername))
+                {
+                    Log.LogError("When using basic HTTP authentication, username cannot be empty.");
+                    return null;
+
+                    // password could be empty
+                }
+
+                var byteArray = Encoding.ASCII.GetBytes($"{BasicAuthenticationUsername}:{BasicAuthenticationPassword}");
```

Comment:
I'm fine with either, especially since I don't really see the point in using basicAuth without a password.